Add lineart preprocessor (#2)
asomoza/image_gen_aux: src/image_gen_aux/preprocessors/README.MD

> # PREPROCESSORS

Reviewer: I think it'd be helpful to add 1-2 sentences describing what preprocessors are and why they're useful so users have a little more context

```suggestion
# Preprocessors
```

A better structure for this doc may be:

```md
# Preprocessors

## LineArtPreprocessor
- usage
- available checkpoints
```

## Supported preprocessors

> This is the list of the currently supported preprocessors

Reviewer: ```suggestion
This is a list of the currently supported preprocessors.
```

* [LineArtPreprocessor](https://github.com/asomoza/image_gen_aux/blob/main/src/image_gen_aux/preprocesors/lineart/README.md)

> ### How to use

Reviewer: Maybe this whole example usage section can be moved under LineArtPreprocessor above. As the list grows, I think that'll give you more space to add more preprocessors and show their usage under it.

Author: @stevhliu What I had in mind is that for the usage of each preprocessor the user has to go to the respective readme under the same directory. This one was more of a generic use for them, but if it's confusing it will be better to just not show an example here?

The only reason I added it here was because all of them are going to have some common args but this also can be done for each of them too.

BTW there's a lot of preprocessors, if we add the usage in this readme it's going to be a long readme, there are some that are different or have more options too, that's why I think it's better to keep the usage example on each one.

Let me know what you think, I'll go with your decision.

Reviewer: Ah I see what you mean! I thought this was a usage example specific to LineArtPreprocessor. But if this is just a generic usage example, then it is totally fine.

I would maybe replace `### How to use` with `### General preprocessor usage` to make it clear its not exclusive to LineArtPreprocessor. 👍

```python
from image_gen_aux import LineArtPreprocessor
from image_gen_aux.utils import load_image

input_image = load_image(
    "https://huggingface.co/datasets/OzzyGT/testing-resources/resolve/main/simple_upscale/hippowaffle.png"
)

lineart_preprocessor = LineArtPreprocessor.from_pretrained("OzzyGT/lineart").to("cuda")
image = lineart_preprocessor(input_image)
image.save("lineart.png")
```

> To load a model you can use the HuggingFace Hub Repository ID, an url or a local path. You can also load `pickle` checkpoints but
> its not recommended. If the checkpoint has custom imports (high risk!!!) you'll need to set the `weights_only` argument to `False`.
> 
> You can also specify a `filename` and a `subfolder` if needed.
> 
> Example:

Reviewer: ```suggestion
Use the Hugging Face Hub model id, a URL, or a local path to load a model. You can also load pickle checkpoints, but it's not recommended due to the [vulnerabilities](https://docs.python.org/3/library/pickle.html) of pickled files. Loading [safetensor](https://hf.co/docs/safetensors/index) files is a more secure option.

If the checkpoint has custom imports (high risk!), you'll need to set the `weights_only` argument to `False`.

You can also specify a `filename` and a `subfolder` if needed.
```

```python
lineart_preprocessor = LineArtPreprocessor.from_pretrained("lllyasviel/Annotators", filename="sk_model.pth", weights_only=False).to("cuda")
```

### List of safetensors checkpoints

> This is the current list of safetensor checkpoints you can use from the hub.

Reviewer: ```suggestion
This is the current list of safetensor checkpoints available on the Hub.
```
